### PR TITLE
feat(github): add pr_state_set tool to correct missed PR state

### PR DESCRIPTION
**Pull request by sh1nj1**
## Problem

A GitHub PR channel's state could only ever change through a `pull_request` webhook. When that delivery was missed — the hook was added after the merge, GitHub failed to deliver, the repo was renamed mid-flight — the chip stayed green on `open` forever, no closing message landed in the topic, and the channel was never detached.

The only user-facing action available was the chip's X (`dismiss!`), which hides the problem rather than fixing it. There was no way to correct the state at all.

## What this adds

`pr_state_set`, an MCP tool that applies the same end-state the webhook would have produced.

| param | required | notes |
|---|---|---|
| `topic_id` | yes | topic the PR channel is attached to |
| `pr_url` | yes | full GitHub PR URL |
| `state` | no | `open` / `merged` / `closed_without_merge` |

**Omitting `state` resynchronizes from the GitHub API**, and that is the preferred form: it structurally cannot record a state the PR does not actually have. `merged` is checked before `state`, because GitHub reports `state: "closed"` for both a merge and an abandonment — reading them the other way round would file every merged PR as `closed_without_merge`.

An explicit `state` remains available as a fallback for repositories with no connected GitHub account, and returns `state_source: "explicit"` so the caller can tell the two apart. When resync is impossible (no account in scope, API call failed) the tool returns `ok: false` with an error that names the fallback rather than guessing.

Scoped to the topic creative and its ancestors — the same RepositoryLink scope `pr_monitor` uses for webhook provisioning, so the tool can never reach a GitHub account the monitor itself could not have used. Before reading a PR, resync verifies that the live repository's stable ID matches a non-null `RepositoryLink#repository_id`, preventing a renamed repository's reused name from targeting the wrong PR. It then tries every identity-valid scoped account until one succeeds. Gated at `authorize_write!`, matching `pr_monitor`.

## Design notes

**`PrChannelStateUpdater` owns the transition.** Closing sets the badge, posts the closing message, then detaches — in that order, so the chip stays visible (now merged/closed) until the user dismisses it. Reopening reactivates, clears `dismissed_at`, and posts the reopened message. The whole transition runs under the channel's row lock in one transaction, so a failed injection rolls the state change back and the chip cannot flash `merged` without the matching timeline message.

**Idempotency is lifecycle-aware, not just state-equality.** A channel can read `open` while sitting detached — the user dismissed the chip, or a reopen event was missed. Treating matching state alone as "already settled" would skip the reactivation half, so `settled?` also checks that the channel's active/detached lifecycle matches the target.

**The webhook path deliberately does not route through the updater.** There, `handle` only *returns* the closing message and `WebhooksController` injects and detaches, because the controller owns the per-channel lock that also de-duplicates GitHub retries. Instead, `closed_message` was extracted as a public method on `GithubPrChannel` and both paths call it — that is the part that would actually drift. A test asserts the manual message is byte-identical to the webhook's.

**Message text is reused verbatim, not distinguished.** A correction is meant to leave the topic reading exactly as it would have if the event had arrived, so recovery is complete rather than annotated. Trade-off: the timeline does not mark that the state was set manually. If an audit marker is wanted later, it is a new i18n key, not a structural change.

**`PrChannelLocator` extracted** so `pr_monitor` and `pr_state_set` resolve a PR URL to a stored channel through identical logic. A mismatch there would mean the corrector silently misses the channel the monitor created. `pr_monitor` was refactored onto it; behavior is unchanged.

## Not in this PR

The chip badge stays a non-interactive `<span>`. The UI for manual state changes is deliberately deferred — `PrChannelStateUpdater` and `PATCH /channels/:id` are the seam it will land on, and `Channel#broadcast_chips_changed` already handles the realtime refresh.

## Tests

- `PrChannelStateUpdaterTest` (12): every transition, both noop shapes, the detached-but-open and merged-but-active edge cases, invalid state, injection-failure rollback, and webhook-message parity.
- `PrStateSetServiceTest` (21): explicit and resync paths, merged-vs-closed disambiguation, stable repository identity verification, reused-name rejection, multi-account fallback, identity lookup failure fallback, canonical-case RepositoryLink, ancestor-creative scope, both fallback errors, mixed-case legacy channel rows, cross-topic isolation, permission denial, tool registration.
- `PrChannelLocatorTest` (6): URL parsing, case-insensitive lookup, ancestor scoping.

All three new files are at **100% line and branch coverage**; every changed line in `github_pr_channel.rb` is covered.

```
engines/collavre_github: 312 runs, 1140 assertions, 0 failures, 0 errors, 0 skips
rubocop: 59 files inspected, no offenses detected
complexity ratchet: OK — 416 entities over budget (417 before), none grew
```

### CODE REVIEW

File: engines/collavre_github/app/services/collavre_github/tools/pr_state_set_service.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      # RepositoryLink keeps GitHub's canonical case.
+      def github_client_for(topic, repo)
+        candidate_ids = scoped_creative_ids(topic)
+        return nil if candidate_ids.empty?
+
+        link = CollavreGithub::RepositoryLink
+          .where("LOWER(repository_full_name) = ?", repo.downcase)
+          .where(creative_id: candidate_ids)
+          .order(:id)
+          .find(&:github_account)
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre_github/app/services/collavre_github/tools/pr_state_set_service.rb]: Fixed in `26f662253`. Resync now collects every identity-valid scoped account in deterministic link order and tries `pull_request_details` until one succeeds. Added coverage where the first account returns `nil`, the second succeeds, and a separate case where an identity lookup error is skipped.

[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre_github/app/services/collavre_github/tools/pr_state_set_service.rb]: Fixed in `26f662253`. Resync now resolves the live repository identity for each scoped account and only accepts an account when that stable ID matches a non-null `RepositoryLink#repository_id`. A reused name with a mismatched ID now fails closed before `pull_request_details` is called. Added a regression test for the reused-name case.

File: engines/collavre_github/app/services/collavre_github/tools/pr_state_set_service.rb [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+        links.group_by(&:github_account_id).filter_map do |account_id, account_links|
+          verified_client_for(account_id, account_links, repo)
+        end
+      end
+
+      def verified_client_for(account_id, links, repo)
+        client = CollavreGithub::Client.new(links.first.github_account)
+        identity = client.repository_identity(repo)
+        return client if links.any? { |link| link.repository_id.to_s == identity.id.to_s }
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre_github/app/services/collavre_github/tools/pr_state_set_service.rb:157]: Fixed in `3f36b56ec`. Resync now applies `RepositoryLink.conflicting_repository_in_scope?` to each live repository identity before accepting a client, matching the webhook fail-closed rule. An ancestor link whose ID matches GitHub can no longer authorize a descendant scope containing the same stored name with a different ID. Added a regression test that verifies the state remains unchanged and `pull_request_details` is never called for this conflict.
